File: src/main/java/ru/xpendence/topjavagraduation/entity/Role.java
package ru.xpendence.topjavagraduation.entity;

import lombok.Getter;
import lombok.Setter;
import ru.xpendence.topjavagraduation.entity.type.RoleType;

import javax.persistence.*;

@Entity
@Table(name = "roles")
@Getter
@Setter
public class Role {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private RoleType name;
}

File: src/main/kotlin/ru/xpendence/topjavagraduation/entity/Role.kt
package ru.xpendence.topjavagraduation.entity

import ru.xpendence.topjavagraduation.entity.type.RoleType

/**
 * Роль.
 */
data class Role(
    val id: Long? = null,
    val name: RoleType? = null,
)
